File: java/src/order/GetOrderEdiDocuments.java
package order;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.ultracart.admin.v2.OrderApi;
import com.ultracart.admin.v2.models.*;
import com.ultracart.admin.v2.util.ApiException;
import java.util.List;

public class GetOrderEdiDocuments {
   /*
       getOrderEdiDocuments returns back all EDI documents associated with an order.

       Possible Errors:
       Order.channelPartnerOid is null -> "Order is not associated with an EDI channel partner."
    */
   public void execute() throws ApiException {
       OrderApi orderApi = new OrderApi(common.Constants.API_KEY);

       String orderId = "DEMO-0009104976";
       OrderEdiDocumentsResponse response = orderApi.getOrderEdiDocuments(orderId);
       List<OrderEdiDocument> documents = response.getEdiDocuments();

       Gson gson = new GsonBuilder().setPrettyPrinting().create();
       for (OrderEdiDocument doc : documents) {
           System.out.println(gson.toJson(doc));
       }
   }
}

File: python/order/get_order_edi_documents.py
from ultracart.apis import OrderApi
from samples import api_client

# get_order_edi_documents() returns all EDI documents associated with an order.
#
# Possible Errors:
# Order.channel_partner_oid is null -> "Order is not associated with an EDI channel partner."

# Create Order API instance
order_api = OrderApi(api_client())

# Order ID to retrieve EDI documents for
order_id = "DEMO-0009104976"

# Retrieve EDI documents
edi_documents = order_api.get_order_edi_documents(order_id).edi_documents

# Print EDI documents
print(edi_documents)
